BHoM_Engine/Modify/SetPropertyValue.cs

```diff
@@ -192,31 +193,74 @@ public static object SetPropertyValue(this object obj, string propName, object v
         /**** Private Methods                           ****/
         /***************************************************/
 
-        private static bool SetValue(this IBHoMObject obj, string propName, object value)
+        private static bool SetValue(this IObject obj, string propName, object value)
         {
-            if (obj == null) return false;
+            if (obj == null || propName == null)
+                return false;
+            
+            // Handle dynamic objects
+            if(obj is IDynamicObject)
+            {
+                if (obj is IDynamicPropertyProvider)
+                {
+                    object result = null;
+                    bool success = Compute.TryRunExtensionMethod(obj, "SetProperty", new object[] { propName, value }, out result);
+
+                    if (success)
+                        return true;
+                }
+                else
+                {
+                    List<PropertyInfo> dynamicProperties = obj.GetType().GetProperties()
+                    .Where(x => x.GetCustomAttribute<DynamicPropertyAttribute>() != null && typeof(IDictionary).IsAssignableFrom(x.PropertyType) && x.PropertyType.GenericTypeArguments.First().IsEnum)
+                    .ToList();
 
-            if (value is IFragment)
+                    // Try to save into a dynamic property
+                    foreach (PropertyInfo prop in dynamicProperties)
+                    {
+                        if (SetValue(prop.GetValue(obj) as dynamic, propName, value))
+                            return true;
+                    }
+                }
+            }
+
+            // Handle BHoM objects
+            if (obj is IBHoMObject)
             {
-                obj.Fragments.Add(value as IFragment);
-                Compute.RecordWarning("The object does not contain any property with the name " + propName + ". The value is being set as a fragment.");
+                SetPropertyFallback(obj as IBHoMObject, propName, value);
+                return true;
             }
             else
             {
-                obj.CustomData[propName] = value;
-                if (!(obj is CustomObject))
-                    Compute.RecordWarning("The object does not contain any property with the name " + propName + ". The value is being set as custom data.");
-            }
+                Compute.RecordWarning("The object does not contain any property with the name " + propName + ".");
+                return false;
+            }                
+        }
+
+        /***************************************************/
 
+        private static bool SetValue<T>(this Dictionary<string, T> dic, string propName, object value)
+        {
+            dic[propName] = (T)(value as dynamic);
```

Comment:
After testing this with my intended approach (using subclasses of Quantities), this casting here does not work for that case. I have a solution working, but might be better with a call to run through that, so lets call when both available!

Reply:
I would personally prefer for the sub-classes of Quantity to implement the operator as I find it much cleaner but I get where you're coming from. So I've added a fix in [this commit].(https://github.com/BHoM/BHoM_Engine/pull/3492/commits/61cff44c2dbb91a571a5599c8c60a1566c7f3347)

Reply:
A bit on the fence on this one. Might be that we are better of reverting that commit, and resort initially to code review, and then adding a new compliance check for subclasses of Quantity<T> to state that they need to implement an explicit cast. Think I prefer that over the activator etc.

Reply:
> Think I prefer that over the activator etc.

Me too

Reply:
Code reverted in this commit: https://github.com/BHoM/BHoM_Engine/pull/3492/commits/b4ff955c8a8a4dfead4c6949107e357e695911e6